numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -39,6 +39,9 @@ NPY_NO_EXPORT PyBoolScalarObject _PyArrayScalar_BoolValues[] = {
     {PyObject_HEAD_INIT(&PyBoolArrType_Type) 1},
 };
 
+/* Maintain bools repr strings, initialized in `initialize_numeric_types` */
+PyObject *_PyBool_ReprValues[2];
```

Comment:
I assume this cache is for performance reasons? Or perhaps to make it the case that `str(np.True_) is str(np.True_)`?

I don't think there's a particularly large benefit to either of these, but if you think they're worthwhile, a comment explaining that this is deliberate would be nice.

Also, to prevent name and linking conflicts:
```suggestion
/* Maintain bools repr strings, initialized in `initialize_numeric_types` */
static PyObject *_PyBoolScalar_ReprValues[2];
```

Reply:
Mainly I was thinking of performance, as we would have to call `PyUnicode_FromString` or something on each repr call. I'll add a comment there and make it static.

Reply:
> as we would have to call PyUnicode_FromString or something on each repr call.

True, although we do way more work in `numpy.void.__repr__` already. For what it's worth, `bool.__repr__` in CPython's `boolobject.c` does go for a similar caching strategy to what you have here.

Reply:
Oh yeah, I like what they are doing by lazily waiting for first call, lets do something similar:
https://github.com/python/cpython/blob/6d883fbe14751b58d9ed2fd708322613d8931035/Objects/boolobject.c#L17-L18

Reply:
Note that the code there is 20 years old - it's possible the optimization no longer makes sense.

Reply:
Oh :) ... Yeah I think lets stick to `return PyUnicode_FromString(s->obval ? "numpy.True_", "numpy.False_");`, much easier to understand and maintain